### ISSUE TITLE
How to write into a PerCpuArray element from eBPF?

### ISSUE THREAD

**Issue by brevilo** [aya-rs/aya#458]
Hi there,

First: thanks for this great project! It's highly appreciated to finally see end-to-end Rust support to write eBPF programs 🚀 

Now, the context for my issue is the `BPF perfbuf` [example](https://nakryiko.com/posts/bpf-ringbuf/#bpf-perfbuf-bpf-perf-event-output) by Andrii Nakryiko where he uses a `BPF_MAP_TYPE_PERCPU_ARRAY` as a heap to get around the eBPF stack size constraint. I need to implement that using Aya as part of a real-world project, but I just can't get it done.

I tried using a single-element `PerCpuArray` for that purpose and using `get_ptr_mut(0)` to obtain write access to that element to change its value (to `42` in the example below), but it doesn't work, despite [being suggested](https://github.com/aya-rs/aya/pull/316#issuecomment-1151232432) otherwise. Whenever I try that the eBPF verifier complains about `R0 invalid mem access 'map_value_or_null'` which means the result of [`bpf_map_lookup_elem()`](https://github.com/aya-rs/aya/blob/main/bpf/aya-bpf/src/maps/per_cpu_array.rs#L70) isn't first compared against `NULL`, at least as far as the verifier is concerned (despite the use of ` NonNull::new()`):

```
291: (85) call bpf_map_lookup_elem#1          ; R0_w=map_value_or_null(id=2,off=0,ks=4,vs=1440,imm=0)
292: (b7) r1 = 42                     ; R1_w=42
293: (73) *(u8 *)(r0 +0) = r1
R0 invalid mem access 'map_value_or_null'
```

tl;dr: does `PerCpuArray` support writing to its elements from within eBPF code at all? If so, how? Maybe my entire approach is wrong?

Thanks in advance!

**Comment by brevilo** [comment]
One potential alternative to my approach could be to use `bpf_map_update_elem()` instead, but that [PR](https://github.com/aya-rs/aya/pull/316) isn't yet merged.

**Comment by brevilo** [comment]
FYI, since the use case here is to store data larger than the stack limit, I could also use the (fairly new) BPF ring buffer, even preferably. Fortunately, there's #294 but it's also still waiting to be merged.

**Comment by alessandrod** [comment]
Hey! Sorry for the late reply - holiday season and all.

Here's an example of how to use `PerCpuArray`https://github.com/alessandrod/aya-echo-tracepoint/blob/main/echo-ebpf/src/main.rst
